### PR TITLE
Rnd 955 support in changing client config parameters

### PR DESCRIPTION
**Pull request by liranbahar**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java [hunk 1]
```diff
@@ -57,7 +57,6 @@ public class AbstractConfig {
 
     /* the original values passed in by the user */
     private final Map<String, ?> originals;
-
```
Comment (shay23b) [inline clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java:60]: return this row -> in future merging from kafka it creates to many diffs

File: clients/src/main/java/org/apache/kafka/common/superstream/Superstream.java [hunk 2]
```diff
@@ -236,41 +235,44 @@ public void registerClient(Map<String, ?> configs) {
...
-                Object learningFactorObject = replyData.get("learning_factor");
-                if (learningFactorObject instanceof Integer) {
-                    learningFactor = (Integer) learningFactorObject;
-                } else if (learningFactorObject instanceof String) {
-                    try {
-                        learningFactor = Integer.parseInt((String) learningFactorObject);
-                    } catch (NumberFormatException e) {
-                        superstreamPrintStream.println(
-                                "superstream: learning_factor is not a valid integer: " + learningFactorObject);
+            if (this.type != null && Arrays.asList(CLIENT_TYPES_LIST).contains(this.type)) {
```
Comment (shay23b) [inline clients/src/main/java/org/apache/kafka/common/superstream/Superstream.java]: At this point we already created a connection to NATS and can create too many unwanted connections. We should check it in abstractConfig (don't even initialize superstream) and for all other places superstream will be null and will not start the other methods/processes

**Review by shay23b — CHANGES_REQUESTED** [review]

File: clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java (hunk 1 above)
Comment (liranbahar) [reply clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java]: done

File: clients/src/main/java/org/apache/kafka/common/superstream/Superstream.java (hunk 2 above)
Comment (liranbahar) [reply clients/src/main/java/org/apache/kafka/common/superstream/Superstream.java]: and done

**Review by shay23b — APPROVED** [review]

**Review by shay23b — APPROVED** [review]

**Pull request merged**